### DISCUSSION TITLE
[Question] How to use "select *" when column is constraint?

### DISCUSSION THREAD

**Discussion by BigStupideer** [trinodb/trino#15126] [Q&A] [answered]
I config access control  policy on a table (name:"test_table") and not allow to visit column  (name:"test_column")
when i submit sql like this:
select * from test_table.
it return access deny error.
how to return other columns in test_table and exclude "test_column"?
i want to get result exclude "test_column" in "select *" query

**Comment by hashhar** [comment, marked as answer]
You can set the config property `hide-inaccessible-columns` to `true` to get that behavior.

It's not the default though since it makes queries non-deterministic (i.e. results depend on who runs them) which is a bad idea in any large system. It's better to be explicit about what you want.

**Reply by BigStupideer** [reply]
get it, thanks
